GravatarApi.php

```diff
@@ -95,4 +112,132 @@ public function exists($email)
         fclose($sock);
         return strpos($header, '404') ? false : true;
     }
+
+    /**
+     * @param Cache $cache
+     */
+    public function setCache(Cache $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    /**
+     * @param int $lifetime
+     */
+    public function setLifetime($lifetime)
+    {
+        $this->lifetime = $lifetime;
+    }
+
+    /**
+     * @param RouterInterface $router
+     */
+    public function setRouter(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * The the URL to access the Gravatar Service
+     *
+     * @param $hash
+     * @param $size
+     * @param $rating
+     * @param $default
+     * @param $secure
+     * @return string
+     */
+    private function generateGravatarServiceUrl($hash, $size, $rating, $default, $secure)
+    {
+        $map = array(
+            's' => $size    ?: $this->defaults['size'],
+            'r' => $rating  ?: $this->defaults['rating'],
+            'd' => $default ?: $this->defaults['default'],
+        );
+
+        $url  = '';
+        $url .= ($secure ? 'https://secure' : 'http://www') . '.gravatar.com/avatar/';
+        $url .= $hash . '?';
+        $url .= http_build_query(
+            array_filter($map)
+        );
+
+        return $url;
+    }
+
+    /**
+     * @param $hash
+     * @param $size
+     * @param $rating
+     * @param $default
+     * @param $secure
+     * @return string
+     */
+    private function generateCachedUrl($hash, $size, $rating, $default, $secure)
+    {
+        if( $this->router === null ) {
+            // Fallback?
+            $url = $this->generateGravatarServiceUrl($hash, $size, $rating, $default, $secure);
+        } else {
+            $url = $this->router->generate(
+                'ornicar_gravatar_image',
+                array(
+                    'hash' => $hash,
+                    'size' => $size ?: $this->defaults['size'],
+                    'rating' => $rating ?: $this->defaults['rating'],
+                    'default' => $default ?: $this->defaults['default'],
+                    'secure' => $secure ? '1' : '0',
+                )
+            );
+        }
+
+        return $url;
```

Comment:
You dont need the else, you can just return from the first condition and have the else be the default.